app/src/main/java/fr/free/nrw/commons/media/MediaClient.java

```diff
@@ -83,6 +83,22 @@ public MediaClient(MediaInterface mediaInterface, MediaDetailInterface mediaDeta
 
     }
 
+    /**
+     * This method takes the userName as input and returns a list of  Media objects filtered using
+     * allimages query It uses the allimages query API to get the images contributed by the userName,
+     * 10 at a time.
+     *
+     * @param userName the username
+     * @return
+     */
+    public Single<List<Media>> getMediaListForUser(String userName) {
+        Map<String, String> continuation =
+            continuationStore.containsKey("user_" + userName) ? continuationStore
+                .get("user_" + userName) : Collections.emptyMap();
```

Comment:
this looks a little funky with the `.get` on the next line, sometimes I favor
```
expression
 ? trueValue
 : falseValue
```

when things run a little long